examples/align/readme.md

```diff
@@ -17,4 +17,186 @@ The application should open a window and display a video stream from the camera.
+
+```cpp
+#include <imgui.h>
+#include "imgui_impl_glfw.h"
+```
+
+These headers are part of the [ImGui](https://github.com/ocornut/imgui) library which we use to render GUI elements. 
+
+Next, we declare three functions to help the code look clearer:
+```cpp
+void render_slider(rect location, float& clipping_dist);
+void remove_background(rs2::video_frame& color, const rs2::depth_frame& depth_frame, float depth_scale, float clipping_dist);
+bool try_get_depth_scale(rs2::pipeline& p, float& scale);
+```
+
+`render_slider(..)`  is where all the GUI code goes, and we will not cover this function in this overview.
+
+`remove_background(..)` takes a depth and color image (that are assumed to be aligned to one another), the depth scale units, and the maximum distance the user wishes to show, and updates the color frame so that its background (any pixel with depth distance larger than the maximum allowed) is remove.
+
+`try_get_depth_scale(..)` is a simple function that tries to find a depth sensor from the pipeline's device and retrieve its depth scale units.
+
+Heading to `main`:
+
+We first define some variables that will be used to show the window and render the images and GUI to the screen:
+
+```cpp
+// Create and initialize GUI related objects
+    window app(1280, 720, "CPP - Align Example"); // Simple window handling
+    ImGui_ImplGlfw_Init(app, false);      // ImGui library intializition 
+    rs2::colorizer c;                          // Helper to colorize depth images
+    texture renderer;                     // Helper for renderig images
+```
+
+Then, we create a context:
+
+```cpp
+// Create a context
+rs2::context ctx;
+```
+`rs2::context` encapsulates all of the devices and sensors, and provides some additional functionalities.
+In this example we will use the `context` to create an `rs2::align` object:
+
+```cpp
+    // Using the context to create a rs2::align object. 
+    // rs2::align allows you to perform aliment of depth frames to others
+    rs2::align align = ctx.create_align(RS2_STREAM_COLOR); 
+```
+
+Next, we define a `rs2::pipeline` to help us easily configure and start the camera.
+
+```cpp
+// Create a pipeline to easily configure and start the camera
+rs2::pipeline pipe;
+```
+
+We start the pipeline using an overloaded version of `start` which takes a callable object.
+
+```cpp
+// By passing align to Pipeline::start, frames will be passed to align's handler
+    pipe.start(align);
+```
+
+`rs2::align` contains an overload for `operator()` which takes a `rs2::frame`. This allows us to pass it as the argument for `pipeline::start`. The pipeline synchronizes its frames before publishing them, thus, by calling `start` with an `align` object, the frames it will provide will be synchronized and aligned to one another. We will soon see how to get these frames from the `align` object.
+
+At this point of the program the camera is started and frames are being aligned.
```

Comment:
the camera ~~is started~~ is configured and streaming and **the resulted** frames are being aligned